cedar-policy-core/src/ast/id.rs

```diff
@@ -63,6 +63,34 @@ impl FromNormalizedStr for Id {
     }
 }
 
+struct IdVisitor;
+
+impl<'de> serde::de::Visitor<'de> for IdVisitor {
```

Comment:
Add a comment that this impl is part of our public API now. Also for the other new impls used by json schema parsing.

As a side note, It'd be good if we could document more thoroughly how each `Serialize` and `Deserialize` implementation gets used. I've found my self wondering a few times if changing the internals of a struct would break part of our API, just DRT, or maybe nothing at all.

Reply:
I don't _think_ that any of the changes in this PR affect our public APIs.
* The `Id` and `Name` types are not publicly exposed, so changes to them should be safe.
* The only type with a `Deserialize` implementation in `cedar-policy` is `PolicyId`, which desugars to `SmolStr` (although possibly we should change that to be `Id`...).
* Most things in `cedar-policy/frontend` are represented directly as strings/JSON. The exceptions are `Decision`, `ValidationEnabled`, and `SchemaFragment`. The first two are not affected by this PR. The last is, but this PR fixes the deserialization to match what it was prior to #749

Reply:
Good point though on better documentation. Once we update DRT to use only public interfaces, we should be able to significantly cut down on the types that support (de)serialization.

Reply:
My point is that this is `Deserialize` impl is now used to define the schema format (part of our api), but until now this `Deserialize` derive here only affected the DRT interface. It'd be easy to make a change here without thinking that it could break the json schema format

Reply:
> although possibly we should change [PolicyId] to be Id

Today, we allow PolicyId's that are any string, not limited to Ids